### PR TITLE
fix(amplify-graphql-types-generator): added functionality for Windows snapshot testing

### PR DESCRIPTION
**Pull request by studpeps**
#### Description of changes
Screenshot 2021-08-05 214742
 \n is a line feed, \r is a carriage return
Different operating systems use different combinations of these characters to represent the end of a line of text
The snapshot test cases have been failing only in Windows, since windows used \\r and \\n together to depict new line escape sequence in this case, which is different from the normal usage of just \n in most other operating systems

Therefore, added a new functionality to replace \\r\\n with \n when being tested on a windows platform

#### Issue #, if available
#7901 

#### Description of how you validated changes
1. cd packages/amplify-graphql-types-generator
2. yarn test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-types-generator/src/loading.ts [hunk 1]
```diff
@@ -46,7 +48,7 @@ function extractDocumentFromJavascript(content: string, tagName: string = 'gql')
 }
 
 export function loadAndMergeQueryDocuments(inputPaths: string[], tagName: string = 'gql'): DocumentNode {
-  const sources = inputPaths
+  var sources = inputPaths
```
Comment (jhockett) [inline packages/amplify-graphql-types-generator/src/loading.ts:51]:
```suggestion
  let sources = inputPaths
```

File: packages/amplify-graphql-types-generator/src/loading.ts [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { buildClientSchema, Source, concatAST, parse, DocumentNode, GraphQLSchem
 import { ToolError } from './errors';
 import { extname, join, normalize, relative } from 'path';
 
+const originalPlatform = process.platform;
```
Comment (jhockett) [inline packages/amplify-graphql-types-generator/src/loading.ts:8]: I don't see the platform being used more than once, so I don't think we need this variable in global scope.

```suggestion
```

File: packages/amplify-graphql-types-generator/src/loading.ts [hunk 3]
```diff
@@ -62,6 +64,13 @@ export function loadAndMergeQueryDocuments(inputPaths: string[], tagName: string
     })
     .filter((source): source is Source => Boolean(source));
 
+    if (originalPlatform === 'win32') {
```
Comment (jhockett) [inline packages/amplify-graphql-types-generator/src/loading.ts:67]: If going with the above suggestion

```suggestion
    if (process.platform === 'win32') {
```

**Comment by cjihrig** [comment]
@phani-srikar, I believe @studpeps is encountering test failures on Windows because of this. Is the replacement package ready to use? Should the existing `amplify-graphql-types-generator` package be removed from this repo, or at least have the tests skipped?

**Comment by phani-srikar** [comment]
> @phani-srikar, I believe @studpeps is encountering test failures on Windows because of this. Is the replacement package ready to use? Should the existing `amplify-graphql-types-generator` package be removed from this repo, or at least have the tests skipped?

Yes. I verified the [Feature Flag deprecation date](https://docs.amplify.aws/cli/reference/feature-flags#useTypesGeneratorPlugin) to remove it. We need to remove the FF from the migrated package and deprecate this one in a separate PR. You can safely ignore those tests for now.

**Comment by danielleadams** [comment]
Hello, this code has since moved to https://github.com/aws-amplify/amplify-codegen. If you are still interested in working on this, please open a pull request in that code base.

**Pull request closed without merging**